src/StringManipulation/NSSolutionTwo.java

```diff
@@ -0,0 +1,50 @@
+class NSSolutionTwo {
+    public boolean backspaceCompare(String S, String T) {
+        
+        int hashCharCount = 0;
+        
+        int len1 = S.length() - 1;
+        int len2 = T.length() - 1;
+        
+        while(len1 >= 0 || len2 >= 0){
+            for(int i = len1; i >= 0; i--){
+                if(S.charAt(i) == '#'){
+                    hashCharCount++;
+                    len1--;
+                }else if(hashCharCount > 0){
+                    hashCharCount--;
+                    len1--;
+                }else{
+                    break;
+                }
+            }
+            
+            for(int j = len2; j >= 0; j--){
+                if(T.charAt(j) == '#'){
+                    hashCharCount++;
+                    len2--;
+                }else if(hashCharCount > 0){
+                    hashCharCount--;
+                    len2--;
+                }else{
+                    break;  
+                }
+            }
+            
+            //checks the character
+            if((len1 >= 0) && (len2 >= 0) && (S.charAt(len1) != T.charAt(len2))){
+                return false;
+            }
+            
+            //checks the string size. At the end they both have to be equal
+            if((len1 != len2) && !((len1 >= 0) && (len2 >= 0))){
```

Comment:
@mvignesh18 the second part of the condition confirms that it is the end. The other simpler way to write this would be len1 < 0 && len2 < 0. I guess the reason he adds this condition is because of the "or" condition he has in the while loop above. Until  both len param becomes zero the loop will carry on. And the below part is where the decrements happen, so I think he is added this check.